pkg/providers/az-client.go

```diff
@@ -23,19 +24,14 @@ type azTenantClient interface {
 	NewListPager(options *armsubscription.TenantsClientListOptions) *runtime.Pager[armsubscription.TenantsClientListResponse]
 }
 
-// GraphServiceClient implements the GraphClient interface.
-type GraphServiceClient struct {
-	Client *msgraph.GraphServiceClient
-}
-
 type GraphClient interface {
-	GetApplicationObjectId(ctx context.Context, appId string) (string, error)
+	Applications() *graphapp.ApplicationsRequestBuilder
```

Comment:
I know we talked about creating an interface out of the `ApplicationRequestBuilder` so that we could mock the `Post` and other calls, are you still working through that or did you decide against it?

Reply:
I tried creating an interface, but the way is it used in other functions, I felt like it may not be a good idea. I can explain in more detail.